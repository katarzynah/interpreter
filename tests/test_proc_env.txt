program f;

var x, y: int;

procedure proc();
var x, z: int;
begin
    x := 5;
    print(x);
    y := y + 1;
    print(y);
    z := x + y;
    print(z);
end;

begin
    x := 0;
    y := 10;
    print(x);
    print(y);
    proc();
    print(x);
    print(y);
end.