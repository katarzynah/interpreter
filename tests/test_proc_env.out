[(fromList [(Ident "x",undefined),(Ident "y",undefined)],fromList [(Ident "proc",ProcDecProc (ProcHead (Ident "proc") (Args ArgListEmpty)) (VarDecFull (VarDecListEnd (VarDecLabel (IdL (Ident "x") (IdLEnd (Ident "z"))) TypeSpecInt))) (CompStmnt (StmntList (SAss (AssStmnt (Ident "x") (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 5))))))) (StmntList (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "x"))))))))))) (StmntList (SAss (AssStmnt (Ident "y") (ExpSimple (SimpleExpAdd (SimpleExpTerm (TermFactor (FactorIdent (Ident "y")))) (TermFactor (FactorConstant (ConstInt 1))))))) (StmntList (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "y"))))))))))) (StmntList (SAss (AssStmnt (Ident "z") (ExpSimple (SimpleExpAdd (SimpleExpTerm (TermFactor (FactorIdent (Ident "x")))) (TermFactor (FactorIdent (Ident "y"))))))) (StmntList (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "z"))))))))))) StmntListEmpty))))))))])]
0
10
5
11
16
0
11
